src/shared/sol-util-file.c

```diff
@@ -86,6 +86,10 @@ sol_util_write_file(const char *path, const char *fmt, ...)
     va_list ap;
     int ret;
 
+    ret = access(path, 0);
+    if (ret < 0)
+        return ret;
+
```

Comment:
I'm afraid that this is not right. `sol_util_write_file` may create a file in some situations, so it isn't right not to allow that. Where is this segfault happening? Could you please send valgrind log of this segfault?
You can get it with:

     valgrind sol-fbp-runner test.fbp &> valgrind.log

Thanks!